codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/endpoints/EndpointMiddlewareGenerator.java

```diff
@@ -121,6 +124,19 @@ private GoWriter.Writable generateAssertRequest() {
                 GoStdlibTypes.Fmt.Errorf);
     }
 
+    private GoWriter.Writable generateAssertRegion() {
+        return goTemplate("""
+                if !$validHost:T(m.options.Region) {
+                    return out, metadata, $error:T("invalid input region %s", m.options.Region)
+                }
+                """,
+                MapUtils.of(
+                        "validHost", SmithyGoTypes.Transport.Http.ValidHostLabel,
```

Comment:
For context, this is the host label function https://github.com/aws/smithy-go/blob/299f4049a6724834ef275ce37d7012e7839dfb99/transport/http/host.go#L73 

it's basically checking `x < 64 && x.match([azAZ09-])`

Reply:
yes, I just call existing host label func used for all endpoint host label check